ios/RNShareActivityItemSource.m

```diff
@@ -0,0 +1,277 @@
+            } else {
+                self->linkMetadata.originalURL = metadata.originalURL;
+                self->linkMetadata.URL = metadata.URL;
+                self->linkMetadata.title = metadata.title;
+                self->linkMetadata.imageProvider = metadata.imageProvider;
+                if (self->linkMetadata.imageProvider) {
+                    self->linkMetadata.iconProvider = self->linkMetadata.imageProvider;
+                } else {
+                    self->linkMetadata.iconProvider = metadata.iconProvider;
+                }
+                self->linkMetadata.remoteVideoURL = metadata.remoteVideoURL;
+                self->linkMetadata.videoProvider = metadata.videoProvider;
+            }
+        }];
+    }
+}
+
+#pragma mark - Utilities
+
++ (BOOL)isURLSchemeData:(nullable NSURL *)URL {
+    return URL && [URL.scheme.lowercaseString isEqual:@"data"];
+}
+
++ (nullable id)URLOrDataFromURL:(nullable NSURL *)URL {
+    if ([self isURLSchemeData:URL]) {
+        NSError *error;
+        return [NSData dataWithContentsOfURL:URL options:(NSDataReadingOptions)0 error:&error];
+    } else {
+        return URL;
+    }
+}
+
++ (nullable id)itemFromDictionary:(NSDictionary *)dictionary {
+    if (dictionary) {
+        NSString *type = [RCTConvert NSString:dictionary[@"type"]];
+        NSString *content = [RCTConvert NSString:dictionary[@"content"]];
+        if ([type isEqual:@"text"]) {
+            return content;
+        }
+        
+        if ([type isEqual:@"url"]) {
+            NSURL *URL = [RCTConvert NSURL:content];
+            return [self URLOrDataFromURL:URL];
+        }
+    }
+    return nil;
+}
+
++ (nullable LPLinkMetadata *)linkMetadataFromDictionary:(NSDictionary *)dictionary API_AVAILABLE(ios(13.0)) {
+    if (dictionary) {
+        LPLinkMetadata *linkMetadata = [[LPLinkMetadata alloc] init];
+        linkMetadata.originalURL = [RCTConvert NSURL:dictionary[@"originalUrl"]];
+        linkMetadata.URL = [RCTConvert NSURL:dictionary[@"url"]];
+        linkMetadata.title = [RCTConvert NSString:dictionary[@"title"]];
+        NSURL *iconURL = [RCTConvert NSURL:dictionary[@"icon"]];
+        if (iconURL) {
+            linkMetadata.iconProvider = [[NSItemProvider alloc] initWithContentsOfURL:iconURL];
+        }
+        NSURL *imageURL = [RCTConvert NSURL:dictionary[@"image"]];
+        if (imageURL) {
+            linkMetadata.imageProvider = [[NSItemProvider alloc] initWithContentsOfURL:imageURL];
+        }
+        linkMetadata.remoteVideoURL = [RCTConvert NSURL:dictionary[@"remoteVideoUrl"]];
+        NSURL *videoURL = [RCTConvert NSURL:dictionary[@"video"]];
+        if (videoURL) {
+            linkMetadata.videoProvider = [[NSItemProvider alloc] initWithContentsOfURL:videoURL];
+        }
+        return linkMetadata;
+    }
+    return nil;
+}
+
++ (nullable NSString *)keyForActivityType:(UIActivityType)activityType {
+    if ([activityType isEqual:UIActivityTypeAddToReadingList]) {
```

Comment:
What do you think on using something like:

```
switch (activityType)
   {
      case isEqual:UIActivityTypeAddToReadingList:
        return @"addToReadingList";
        break;
}
```

Reply:
@MateusAndrade Sorry for my late reply. `switch` statement doesn't work with string in Objective-C. There are some ways to do equivalent, for example, using macros or using dictionary of blocks, but I think they are not cool. Anyway, thank you for merge!